mast_aladin_lite/overlay_manager.py

```diff
@@ -0,0 +1,96 @@
+import warnings
+from mast_aladin_lite.mast_overlay import MastOverlay
+
+
+class OverlayManager:
+    def __init__(self):
+        self._overlays_dict = {}
+
+    def __setitem__(self, key, value):
+        self._overlays_dict[key] = value
+
+    def __getitem__(self, key):
+        return self._overlays_dict[key]
+
+    def __contains__(self, key):
+        return key in self._overlays_dict
+
+    def items(self):
+        return self._overlays_dict.items()
+
+    def pop(self, key):
+        self._overlays_dict.pop(key)
+
+    def keys(self):
+        return self._overlays_dict.keys()
+
+    def make_unique_name(self, name):
+        """Create a unique layer name.
+
+        Parameters
+        ----------
+        name : str
+            The current name of the layer to be added to the widget.
+
+        Returns
+        -------
+        unique_name
+            A string that is a unique name for the layer being added.
+        """
+        unique_name = name
+        i = 1
+
+        while unique_name in self._overlays_dict:
+            unique_name = f"{name}_{i}"
+            i += 1
```

Comment:
nit: In the worst case, this loop is O(m*n) where m is the number of duplicate names (overlay_1, overlay_2, ..., overlay_m) and n is the number of overlays in the dictionary. We _could_ speed this up, but I don't think its that important right now. Something to consider if we ever have overlay power users that have 100's of overlays with the same name